Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -207,121 +203,92 @@ class SentryProfilerSwiftTests: XCTestCase {
 
     private var fixture: Fixture!
 
+    override class func setUp() {
+        super.setUp()
+        SentryLog.configure(true, diagnosticLevel: .debug)
+    }
+
     override func setUp() {
         super.setUp()
         fixture = Fixture()
-        SentryLog.configure(true, diagnosticLevel: .debug)
-        CurrentDate.setCurrentDateProvider(fixture.currentDateProvider)
-        fixture.options.profilesSampleRate = 1.0
-        fixture.options.tracesSampleRate = 1.0
-        fixture.prepareMetricsMocks()
     }
 
     override func tearDown() {
         super.tearDown()
+        fixture.timeoutTimerWrapper.fire()
```

Comment:
If a test early exits because of a thrown error, it may not finish the spans it created. This ensures any open spans are closed before starting the next test case.

Reply:
Please add this info as a comment.

Reply:
i also added a function to clear the bookkeeping array in SentryTracerConcurrency